org.jacoco.core/src/org/jacoco/core/internal/analysis/MethodAnalyzer.java

```diff
@@ -355,8 +356,16 @@ private void visitSwitchTarget(final Label label, final int branch) {
 	public void visitEnd() {
 		// Wire jumps:
 		for (final Jump j : jumps) {
-			LabelInfo.getInstruction(j.target).setPredecessor(j.source,
-					j.branch);
+
+			final Instruction target = LabelInfo.getInstruction(j.target);
+			final int sourceOpcode = j.source.getNode().getOpcode();
+			if ((sourceOpcode == Opcodes.LOOKUPSWITCH
+					|| sourceOpcode == Opcodes.TABLESWITCH)
+					&& ignored.contains(target.getNode())) {
```

Comment:
Better to make this explicit.

Reply:
It looks like we don't have unit tests in `MethodAnalyzerTest` which covers this?

Reply:
As was discussed by phone: this even should not be here - filter should play role during counting and should not change CFG construction.

Reply:
I believe that this can be replaced by API that is being developed in #735